### ISSUE TITLE
Analyze if a Retry combinator could be useful

### ISSUE THREAD

**Issue by danieldietrich** [vavr-io/vavr#1568]
Failing computations (Future, Try, ...) could be retried.

Haskell has a Retry combinator, see the [package docs](https://hackage.haskell.org/package/retry) and the [repo](https://github.com/Soostone/retry).

**Goal:**

Before we start to implement a `Retry` control or dive too deep into implementation details we need to create some code example from the user perspective (API).
- How do use-cases look like, e.g. reading an entity from a database using retry-logic e.g. if the network is down?

**Comment by danieldietrich** [comment]
Out of scope for Vavr
